services/frontend-v3/src/components/searchBar/SearchBar.jsx

```diff
@@ -48,14 +45,56 @@ const SearchBar = () => {
     setClassOptions(results.data);
   };
 
+  const handleAnyMentorSkillsChange = (e) => {
+    setAnyMentorSkills(e.target.checked);
+  };
+
   // turns search values into query, redirects to results page with query
-  const handleSearch = (values) => {
+  const handleSearch = (unassignableValues) => {
+    let values;
+    if (unassignableValues.anyMentorSkills) {
+      values = { ...unassignableValues, mentorshipSkills: undefined };
+    } else {
+      values = { ...unassignableValues, anyMentorSkills: undefined };
+    }
     const query = queryString.stringify(values, { arrayFormat: "bracket" });
     const url = `/secured/results?${encodeURI(query)}`;
     history.push(url);
   };
 
   useEffect(() => {
+    // Fetches options for skills select field in advanced search
+    const getSkills = async () => {
+      const dataTree = [];
+
+      // Get user profile
+      const url = `${backendAddress}api/option/getCategory`;
+      const result = await axios.get(url);
+
+      // Loop through all skill categories
+      for (let i = 0; i < result.data.length; i += 1) {
+        const parent = {
+          title: result.data[i].description[locale],
+          value: result.data[i].id,
+          children: [],
+        };
+
+        dataTree.push(parent);
+        // Loop through skills in each category
+        for (let w = 0; w < result.data[i].skills.length; w += 1) {
+          const child = {
+            title: `${result.data[i].description[locale]}: ${result.data[i].skills[w].description[locale]}`,
+            value: result.data[i].skills[w].id,
+            key: result.data[i].skills[w].id,
+          };
+          dataTree[i].children.push(child);
+        }
+      }
+
+      setSkillOptions(dataTree);
+      return 1;
+    };
+
```

Comment:
Nevermind, you'll also need to update this, not exactly sure what endpoint replaces `api/option/getCategory`, maybe `api/option/categories`, but it just returns the categories, with no link to the skills. I think you need to query `api/option/skills` which returns the skills with the associated categoryId, then make another query to `api/option/categories`, then loop over the skills and get the appropriate category name for each skill